src/prerequisites/signals.py

```diff
@@ -87,3 +88,33 @@ def update_cache_triggered_by_prereq(sender, instance, *args, **kwargs):
         # Cover this instance in a post_delete signal receiver for Quest objects.
         if instance.parent_object:
             update_conditions_for_quest.apply_async(kwargs={'quest_id': instance.parent_object.id, 'start_from_user_id': 1}, queue='default')
+
+
+@receiver(post_save, sender=Prereq)
+def on_quest_badge_save_with_rank_prereq(sender, instance, *args, **kwargs):
+    """ This function checks if rank was added as a prerequisite and doesn't exist as an initial object of CytoScape map,
+    then creates a new map.
+    """
+    parent_model_class = instance.parent_content_type.model_class()
+    if not issubclass(parent_model_class, HasPrereqsMixin):
+        return
+
+    # check if preq 'slot' (prereq/or_prereq) has the conditions needed to create Cyto map
+    # create map if so
+    def check_conditions_and_create_map(prereq_invert, prereq_content_type, prereq_object_id):
```

Comment:
Move comment above method into a docstring and properly document in docstring, all parameters and what this does